### ISSUE TITLE
Support multiple lines with a crosshair

### ISSUE THREAD

**Issue by zeleniy** [d3fc/d3fc#821]
Please see this example of nvd3.js library - http://nvd3.org/examples/line.html. Here vertical line shows intersection with each line on the chart and tool tip shows aggregated iformation.
Does it possible to to the same with [crosshair](http://d3fc.io/components/tool/crosshair.html) component?

**Comment by ColinEberhardt** [comment]
Very good question. The crosshair itself supports rendering a single point with vertical / horizontal annotations. However, there is no reason why you couldn't use multiple crosshairs joined together.

I'll see if I can come up with an example of this at some point later today.

**Comment by zeleniy** [comment]
Colin thank you for participating. Maybe you already have an answer? Or you can prompt me what and how to modify?

**Comment by rymdluo** [comment]
I would be interested in looking at such an example as well @ColinEberhardt !

**Comment by ColinEberhardt** [comment]
OK, had a bit of time to play around with this. While the crosshair itself doesn't natively support multiple series, d3fc is based on the concept of small components, and as a result you can construct this behaviour yourself by using some of the more 'low level' components and features.

The d3fc crosshair is bound (via D3 data-join) to an array which is populated with a single item that represents the current crosshair location. The crosshair is visually constructed using a point series and two line annotations.

The snapping concept is also a separate piece of functionality. 

You can use the component parts to create the required result, a crosshair that tracks multipel series.

Here's an example:

https://jsbin.com/jepoverovu/edit?js,output

What I have done is construct data that is bound to the chart, which has two arrays, one for the 'real' crosshair (which tracks the sine curve) and one for the point that tracks the other series.

The tracking events (e.g. `trackingmove`) for the crosshair are handled and here the snapping function is applied to the cosine series. This updates the array which is 'bound' to the point that tracks the cosine series.

Notice that the chart is re-rendered in its entirety each time. This is a pattern which we favour, letting D3 take care of optimizing the re-render.

Whilst this works, it isn't ideal. I'll take a look at whether the crosshair can be enhanced to more 'natively' support multiple series.

Hope that helps!

**Comment by zeleniy** [comment]
Ok, thank you. But how to show tooltip with aggregated information from both lines?

**Comment by ColinEberhardt** [comment]
Having tooltips for the aggregated information would be tricky - as I mentioned, I think crosshair itself should be enhanced to support this. I've changed the issue title to reflect this.

**Comment by zeleniy** [comment]
If i want to implement my own crosshair how i can append "mousemove" event handler? Something like:
```
var multi = fc.series.multi()
  .series([gridlines, sinLine, cosLine, point, crosshair])
  .mapping(function(series) {
      switch (series) {
          case crosshair:
              return data.sinCrosshair;
          case point:
              return data.cosCrosshair;
          default:
              return data.series;
      }
  }).on("mousemove", function(d, i) {
    ...
  })
```

**Comment by chrisprice** [comment]
Fixed by #999.
